azext_iot/_params.py

```diff
@@ -39,7 +39,23 @@
 from azure.cli.core.local_context import LocalContextAttribute, LocalContextAction
 
 
-auth_type_dataplane_param_type = CLIArgumentType(
+dps_auth_type_dataplane_param_type = CLIArgumentType(
+    options_list=["--auth-type"],
+    arg_type=get_enum_type(
+        AuthenticationTypeDataplane, AuthenticationTypeDataplane.key.value
+    ),
+    arg_group="Access Control",
+    help="Indicates whether the operation should auto-derive a policy key or use the current Azure AD session. "
+    "You can configure the default using `az configure --defaults iotdps-data-auth-type=<auth-type-value>`",
+    configured_default="iotdps-data-auth-type",
+    local_context_attribute=LocalContextAttribute(
```

Comment:
Lets remove usage of local_context_attribute here and for hub_auth_type_dataplane_param_type. It is still experimental functionality and we haven't advertised it.

Reply:
It's been there for a while - I would rather keep both local context attributes (at least the hub one)

There may be users using the local hub variable because the variable has been there. It makes more sense to separate the hub and dps variables (especially as the dps rbac is a newer feature) to let users more granular control. Thus, I lean heavily on both being present.

Reply:
local context in az CLI is both deprecated and experimental 
![image](https://user-images.githubusercontent.com/28658112/145458742-e7fcc524-b495-4420-ac74-783dd925c0fd.png)

and is independent from az cli configuration.

Reply:
I also don't understand this comment 

"It makes more sense to separate the hub and dps variables (especially as the dps rbac is a newer feature) to let users more granular control. "

As this feedback item is not making an assertion against separate hub and dps variables.

Reply:
removed.